tests/jlm/llvm/frontend/llvm/LlvmTypeConversionTests.cpp

```diff
@@ -0,0 +1,110 @@
+    llvm::Type * llvm_type,
+    jlm::llvm::fpsize jlm_type_size)
+{
+  using namespace llvm;
+
+  std::unique_ptr<Module> llvmModule(new Module("module", context));
+
+  auto pointer_type = ::llvm::PointerType::getUnqual(llvm_type);
+
+  auto function_type = FunctionType::get(llvm_type, ArrayRef<Type *>({ pointer_type }), false);
+  auto function =
+      Function::Create(function_type, GlobalValue::ExternalLinkage, "f", llvmModule.get());
+
+  auto bb = BasicBlock::Create(context, "bb", function);
+
+  auto arg0 = function->getArg(0);
+
+  IRBuilder<> builder(bb);
+
+  auto loadedValue1 = builder.CreateLoad(llvm_type, arg0, false);
+  builder.CreateRet(loadedValue1);
+
+  // Act
+  auto ipgModule = jlm::llvm::ConvertLlvmModule(*llvmModule);
+  print(*ipgModule, stdout);
+
+  {
+    using namespace jlm::llvm;
+    auto controlFlowGraph =
+        dynamic_cast<const function_node *>(ipgModule->ipgraph().find("f"))->cfg();
+    auto basicBlock =
+        dynamic_cast<const basic_block *>(controlFlowGraph->entry()->outedge(0)->sink());
+
+    /*     Generated ipg bb looks something like this:
+        <double, iostate, mem> f <ptr, iostate, mem>
+        {
+        0xbd0675417b80:  _io_ _s_
+
+        0xbd0675418110:
+          _r_ = undef
+          tv0, tv1 = Load , _s_
+          ASSIGN _s_, tv1
+          ASSIGN _r_, tv0
+          [0xbd0675417c00]
+
+        0xbd0675417c00: _r_ _io_ _s_
+        } */
+
+    auto tac_it = basicBlock->begin();
+    auto return_declaration = *tac_it;
+    auto load = *std::next(tac_it);
+    auto return_assignment = *std::next(tac_it, 3);
+
+    std::cout << return_assignment->operation().debug_string() << std::endl;
+
+    assert(is<fptype>(return_declaration->result(0)->type()));
+    auto t0 = dynamic_cast<const fptype &>(return_declaration->result(0)->type());
+    assert(t0.size() == jlm_type_size);
+
+    assert(is<fptype>(load->result(0)->type()));
+    auto t1 = dynamic_cast<const fptype &>(load->result(0)->type());
+    assert(t1.size() == jlm_type_size);
+
+    assert(is<fptype>(return_assignment->operand(0)->type()));
+    auto t2 = dynamic_cast<const fptype &>(return_assignment->operand(0)->type());
+    assert(t2.size() == jlm_type_size);
+
+    assert(is<fptype>(return_assignment->operand(1)->type()));
+    auto t3 = dynamic_cast<const fptype &>(return_assignment->operand(1)->type());
+    assert(t3.size() == jlm_type_size);
+  }
+}
+
+static int
+TypeConversion()
+{
+  llvm::LLVMContext context;
+
+  TestTypeConversion(context, ::llvm::Type::getHalfTy(context), jlm::llvm::fpsize::half);
+  TestTypeConversion(context, ::llvm::Type::getFloatTy(context), jlm::llvm::fpsize::flt);
+  TestTypeConversion(context, ::llvm::Type::getDoubleTy(context), jlm::llvm::fpsize::dbl);
+  TestTypeConversion(context, ::llvm::Type::getX86_FP80Ty(context), jlm::llvm::fpsize::x86fp80);
+  TestTypeConversion(context, ::llvm::Type::getFP128Ty(context), jlm::llvm::fpsize::fp128);
+
+  return 0;
+}
```

Comment:
I appreciate that you wrote the test, but this does not only test the type conversion, but a lot of other stuff with it as well. It should really only test the type conversion itself:

```
#include <jlm/llvm/frontend/LlvmTypeConversion.hpp>

auto llvmType = ::llvm::Type::getHalfTy(context);
... // setup other LLVM types

auto jlmType = jlm::llvm::ConvertType(llvmType, ...);
... // convert other LLVM types

auto floatingPointType = dynamic_cast<const fptype*>(jlmType.get());
assert(floatingPointType && floatingPointeType->size() == fpsize::half);
... // assert other LLVM type conversions
```

The same pattern for the backend type conversion.

You can skip the tests for the instruction conversion (this is covered by the LLVM test suite tests etc.).

Reply:
I have made the requested changes, but I did not quite understand which conversions I should test, did you only want them for the floating point types?

Reply:
Looks good. You only touched the floating point types, so it is fine that you only add tests for those. Thanks a lot!